### PR TITLE
ww3 spec

### PR DESCRIPTION
**Pull request by aliabdolali**
# Pull Request Summary
add three matlab functions:

write_directional_spectra_ascii.m
write_directional_spectra_nc.m 
convert_time.m 

## Description
write_directional_spectra_ascii.m to write directional spectral density file in WW3 format (ascii)
write_directional_spectra_nc.m to write directional spectral density file in WW3 format (netcdf)
convert_time.m to read time from netcdf and convert to matlab time (read reference date and unit from time variable and adjust)

Provide a detailed description of what this PR does.
new feature, to provide an option for the users to generate boundary conditions for WW3 model.

Please also include the following information: 
_enhancement_, _new feature_

### Commit Message
add functions to write directional spectral density files in WW3 format (ascii and nc)

### Check list  

- [x] Branch is up to date with the authoritative repository (NOAA-EMC) develop branch. 
 
### Testing

* How were these changes tested?
These funtions have been tested for Boes cases and wave vegetation cases in WW3.

### CODE REVIEW

File: tools/convert_time.m [hunk 1]
```diff
@@ -0,0 +1,36 @@
+function [dates]=convert_time(ncf,varid)
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% This program cobverts the time variable in the netcdf file relative to  %
```
Comment (MatthewMasarik-NOAA) [inline tools/convert_time.m]: converts

File: tools/convert_time.m [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+% unit origin (unit attrivute)to matlab time                              %
```
Comment (MatthewMasarik-NOAA) [inline tools/convert_time.m]: attribute

**Review by MatthewMasarik-NOAA — CHANGES_REQUESTED** [review]
First commit for 3 matlab `tools/` scripts, developer has tested.  Code review requests correction of two typos, otherwise passes.

**Comment by MatthewMasarik-NOAA** [comment]
@aliabdolali please edit typos then I will merge this PR.

**Comment by aliabdolali** [comment]
> @aliabdolali please edit typos then I will merge this PR.

Done!!!!

**Review by MatthewMasarik-NOAA — APPROVED** [review]
Thank you @aliabdolali for contributing these utility scripts for the community to use.

**Pull request merged**